packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/PersistedNameTests.kt

```diff
@@ -0,0 +1,312 @@
+        assertTrue { migrationTriggered }
+    }
+
+    @Test
+    fun schema_changingPublicName_doesNotTriggerMigration() {
+        val oldSchema = setOf(io.realm.kotlin.entities.migration.before.PublicNameChangeMigrationSample::class)
+        val newSchema = setOf(io.realm.kotlin.entities.migration.after.PublicNameChangeMigrationSample::class)
+
+        // Open a realm with the old schema
+        val oldConfig = RealmConfiguration
+            .Builder(schema = oldSchema)
+            .name("foo.realm")
+            .directory("$tmpDir/bar")
+            .build()
+        val oldRealm = Realm.open(oldConfig)
+
+        // Add an object to the realm and close it
+        oldRealm.writeBlocking {
+            copyToRealm(io.realm.kotlin.entities.migration.before.PublicNameChangeMigrationSample())
+        }
+        oldRealm.close()
+
+        // Open a realm with the new schema
+        val newConfig = RealmConfiguration
+            .Builder(schema = newSchema)
+            .name("foo.realm")
+            .directory("$tmpDir/bar")
+            .migration(
+                AutomaticSchemaMigration {
+                    fail("Migration triggered")
+                }
+            )
+            .build()
+
+        // Migration should not be needed
+        Realm.open(newConfig).close()
+    }
+
+    private fun <T> assertCanQuerySingle(property: KMutableProperty1<PersistedNameSample, T>, nameToQueryBy: String, value: T) {
+        realm.query<PersistedNameSample>("$nameToQueryBy = $0", value)
+            .find()
+            .single()
+            .run {
+                assertNotNull(this)
+                assertEquals(value, property.getValue(this, property))
+            }
+    }
+}
+
+class PersistedNameSample : RealmObject {
+
+    @PersistedName("persistedNamePrimaryKey")
+    @PrimaryKey
+    var publicNamePrimaryKey: BsonObjectId = bsonObjectId
+
+    @PersistedName("persistedNameStringField")
+    var publicNameStringField: String = "Realm"
+
+    @PersistedName("persistedNameTimestampField")
+    var publicNameTimestampField: RealmInstant = RealmInstant.from(100, 1000)
+
+    @PersistedName("persistedNameObjectIdField")
+    var publicNameObjectIdField: ObjectId = objectId
+
+    @PersistedName("persistedNameBsonObjectIdField")
+    var publicNameBsonObjectIdField: BsonObjectId = bsonObjectId
+
+    @PersistedName("persistedNameUuidField")
+    var publicNameUuidField: RealmUUID = RealmUUID.from("46423f1b-ce3e-4a7e-812f-004cf9c42d76")
+
+    @PersistedName("persistedNameBinaryField")
+    var publicNameBinaryField: ByteArray = byteArrayOf(42)
+
+    @PersistedName("persistedNameStringListField")
+    var publicNameStringListField: RealmList<String> = realmListOf()
+
+    @PersistedName("persistedNameStringSetField")
+    var publicNameStringSetField: RealmSet<String> = realmSetOf()
+
+    @PersistedName("persistedNameChildField")
+    var publicNameChild: PersistedNameSample? = null
+}
```

Comment:
Which of these remaining fields do you think we should keep for testing purposes? (i.e. we could perhaps remove some)